pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/FixedByteSingleValueMultiColWriter.java

```diff
@@ -38,11 +38,11 @@ public FixedByteSingleValueMultiColWriter(File file, int rows, int cols,
     this.columnOffsets = new int[cols];
     rowSizeInBytes = 0;
     for (int i = 0; i < columnSizes.length; i++) {
-      columnOffsets[i] = rowSizeInBytes;
+      columnOffsets[i] = (int) rowSizeInBytes;
```

Comment:
I am probably missing something here: 'rowSizeInBytes' is changed to 'long', but then is being casted to 'int' in a lot of places. Given that it was explicitly changed to 'long' implies you are expecting size that would not fit in it. If so, can it be safely casted to 'int' then?